core/pom.xml

```diff
@@ -163,6 +163,7 @@ limitations under the License.
           <includes>
             <include>org/apache/calcite/test/CalciteSuite.java</include>
           </includes>
+          <argLine>-Xmx1536m -XX:MaxPermSize=256m -Duser.timezone=UTC</argLine>
         </configuration>
```

Comment:
I copied this piece of code from pom.xml in the root of the project, and I changed the timezone.
- Xmx has the same value associated as in the root pom.
- I agree we could get rid of MaxPermSize, as it is deprecated in 1.8.
- The purpose of copying that line was indeed changing the timezone to UTC for the tests. Otherwise, DateRangeRules and RelOptRulesTest(testExtractYearMonthToRange) are not passing in my machine (BST); this is independent of this patch.

E.g.
  DateRangeRulesTest.testExtractYearAndMonthFromDateColumn:90->checkDateRange:182
Expected: is "AND(AND(>=($8, 2014-01-01), <($8, 2015-01-01)), AND(>=($8, 2014-06-01), <($8, 2014-07-01)))"
     but: was "AND(AND(>=($8, 2014-01-01), <($8, 2015-01-01)), AND(>=($8, 2014-05-31), <($8, 2014-06-30)))"

In fact, the root pom sets the timezone to the user timezone. Calendar objects store information about the timezone. When we print the date (RexLiteral.printDatetime) in the test results, we take the millisecond value, but we do not perform any conversion...

This is somehow related to the point you raised in the JIRA issue.

A proposal could be:
1) Create the Calendar object in DateRangeRules in UTC (instead of default user timezone as we do now). That solves the issue with the tests described above.
2) Use UTC for Intervals sent to Druid too, as using user timezone in Calcite when generating Druid intervals will only lead to confusion.

What do you think?